bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -0,0 +1,503 @@
+                logger.error("JsonException: {}", e);
+            }
+        } else if (jsonResponse.contains("error")) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Error communicating with Tibber API");
+            return;
+
+        } else {
+            logger.debug("Unexpected response from Tibber: {}", jsonResponse);
+            return;
+        }
+    }
+
+    public void startRefresh(int Refresh) {
+        if (pollingJob == null) {
+            pollingJob = scheduler.scheduleWithFixedDelay(() -> {
+                try {
+                    updateRequest();
+                } catch (IOException e) {
+                    logger.error("IO Exception: {}", e);
+                }
+            }, 1, Refresh, TimeUnit.MINUTES);
+        }
+        return;
+    }
+
+    public void updateRequest() throws IOException {
+        getURLPriceInfo(BASE_URL);
+        getURLDaily(BASE_URL);
+        getURLHourly(BASE_URL);
+
+        if (rtEnabled.equals("true") && !isConnected()) {
+            try {
+                open();
+            } catch (Exception e) {
+                logger.error("Exception: {}", e);
+            }
+        }
+
+        return;
+    }
+
+    public void updateChannel(String channelID, String channelValue) {
+        if (!channelValue.contains("null")) {
+            BigDecimal state = new BigDecimal(channelValue);
+            updateState(channelID, new DecimalType(state));
+            return;
+        } else {
+            return;
+        }
+    }
+
+    public void thingStatusChanged(ThingStatusInfo thingStatusInfo) {
+        logger.warn("Thing Status updated to {} for device: {}", thingStatusInfo.getStatus(), getThing().getUID());
+        if (thingStatusInfo.getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Unable to communicate with Tibber API");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+        }
+        if (isConnected()) {
+            close();
+        }
+        super.dispose();
+    }
+
+    public void open() throws Exception {
+        if (isConnected()) {
+            logger.warn("Open: connection is already open");
+        }
+        logger.info("Connecting to: {}", SUBSCRIPTION_URL);
+
+        sslContextFactory = new SslContextFactory(true);
+        sslContextFactory.setTrustAll(true);
+        sslContextFactory.setEndpointIdentificationAlgorithm(null);
+
+        client = new WebSocketClient(sslContextFactory);
+        client.setMaxIdleTimeout(360 * 1000);
+
+        socket = new TibberWebSocketListener();
+
+        request = new ClientUpgradeRequest();
+        String token = new StringBuilder("Bearer ").append(configuration.getToken()).toString();
+        request.setHeader("Authorization", token);
```

Comment:
I had quite a struggle to get this one right, and have tried with the suggested change before. 

As can be seen from [https://community.openhab.org/t/solved-struggling-to-set-up-oauth-token-bearer-in-new-binding/85181/11](https://community.openhab.org/t/solved-struggling-to-set-up-oauth-token-bearer-in-new-binding/85181/11) the `StringBuilder.append` was the way to go to fix my problem.

But this might be a structural issue with use of variables you point out, so maybe I could change to:
`request.setHeader("Authorization", new StringBuilder("Bearer").append(configuration.getToken()).toString());`

Reply:
I had quite a struggle to get this one right, and have tried with the suggested change before. 

As can be seen from [https://community.openhab.org/t/solved-struggling-to-set-up-oauth-token-bearer-in-new-binding/85181/11](https://community.openhab.org/t/solved-struggling-to-set-up-oauth-token-bearer-in-new-binding/85181/11) the `StringBuilder.append` was the way to go to fix my problem.

But this might be a structural issue with use of variables you point out, so maybe I could change to:
`request.setHeader("Authorization", new StringBuilder("Bearer").append(configuration.getToken()).toString());`